Google Script (JS)

```diff
@@ -0,0 +1,57 @@
+function doPost(e) {
+  
+  var isProduction = false;
+  
+  //if(typeof e == 'undefined')
+    //return ContentService.createTextOutput(JSON.stringify(e.parameter));
+  
+  var strSimulator  = "https://www.sandbox.paypal.com/cgi-bin/webscr"
+  var strLive = "https://www.paypal.com/cgi-bin/webscr"
+  var paypalURL = strSimulator
+  
+  if (isProduction)  paypalURL = strLive;
+  var payload = "cmd=_notify-validate&" + e.postData.contents;
+  payload = payload.replace("+", "%2B");
+
+  var options =
+    {
+      "method" : "post",
+      "payload" : payload,
+    };
+    
+  var resp = UrlFetchApp.fetch(paypalURL, options); //Handshake with PayPal - send acknowledgement and get VERIFIED or INVALID response
+  
+  if (resp == 'VERIFIED') {
+    if (e.parameter.payment_status  == 'Completed') {
+      if (e.parameter.receiver_email == 'receiver@email.com') {
+        //Convert to reference currency (USD) if paid in any other currency
```

Comment:
For the purposes of the example, I think it's best to remove this conversion logic. I would replace it with a conditional check the that the currency is correct.